pkg/apis/core/v1alpha1/registry_types_test.go

```diff
@@ -0,0 +1,62 @@
+package v1alpha1_test
```

Comment:
I moved these tests from `internal/container/default_registry_test.go`

These are the first _explicit_ tests we have for `Validate()` methods on API types (😅). I think it makes sense for them to live here along with the types, but calling it out because there's currently no other `_test.go` files here and the API types package is kind of special with all the codgen stuff